### PR TITLE
Add Contextual Menu Actions

### PR DESCRIPTION
**Pull request by gianfilippocisternino**
Add contextual menu actions based on the node kind.

### CODE REVIEW

File: Sources/App/Models/FileSystemNode.swift [hunk 1]
```diff
@@ -47,6 +45,17 @@ struct FileSystemNode: Identifiable, Hashable {
...
 
+  /// The list of available actions that can be performed on the node.
+  var availableActions: [Action] {
+    switch kind {
+    case let .folder(_, isRequestFolder: isRequestFolder):
+      return isRequestFolder ? [.edit, .delete] : [.create]
+
+    case .requestFile:
+      return [.edit, .delete]
+    }
```
Comment (TheInkedEngineer) [inline Sources/App/Models/FileSystemNode.swift:56]:
```suggestion
  /// The list of available actions that can be performed on the node.
  var availableActions: [Action] {
    switch kind {
    case let .folder(_, isRequestFolder):
      return isRequestFolder ? [.edit, .delete] : [.create]

    case .requestFile:
      return [.edit, .delete]
    }
```

File: Sources/App/Views/Sections/Editor/Detail/EditorDetailViewModel.swift [hunk 2]
```diff
@@ -177,14 +180,17 @@ final class EditorDetailViewModel: ObservableObject {
...
   ///   - completion: A closure to invoke when the user taps the `Save` button.
+  ///   - onCancel: A closure to invoke when the user taps the `Cancel` button.
   init(
     requestFile: FileSystemNode? = nil,
     requestFolder: FileSystemNode? = nil,
     requestParentFolder: FileSystemNode? = nil,
     mode: Mode = .read,
-    onSave completion: Interaction? = nil
+    onSave completion: Interaction? = nil,
+    onCancel: Interaction? = nil
```
Comment (TheInkedEngineer) [inline Sources/App/Views/Sections/Editor/Detail/EditorDetailViewModel.swift:190]: At this point let's remove completion.
```suggestion
    onSave: Interaction? = nil,
    onCancel: Interaction? = nil
```

update the init and the doc.

File: Sources/App/Views/Sections/Editor/Detail/EditorDetailViewModel.swift [hunk 3]
```diff
@@ -207,10 +213,13 @@ final class EditorDetailViewModel: ObservableObject {
       isResponseBodyTextFieldEnabled = false
     }
 
-    guard case let .requestFile(request) = requestFile?.kind, let requestFolder = requestFolder, let requestParentFolder = requestParentFolder else {
+    guard case let .requestFile(request) = requestFile?.kind, let requestFolder = requestFolder,
+      let unwrappedRequestParentFolder = requestParentFolder
+    else {
```
Comment (TheInkedEngineer) [inline Sources/App/Views/Sections/Editor/Detail/EditorDetailViewModel.swift:218]:
```suggestion
    guard 
      case let .requestFile(request) = requestFile?.kind, 
      let requestFolder = requestFolder,
      let unwrappedRequestParentFolder = requestParentFolder
    else {
```

File: Sources/App/Views/Sections/Editor/SourceTree/SourceTree.swift [hunk 4]
```diff
@@ -30,8 +30,13 @@ struct SourceTree: View {
...
               menuItems: {
-                Button("Delete", action: {})
-                Button("Edit", action: {})
+                ForEach(node.availableActions, id: \.self) { action in
+                  Button(
+                    viewModel.actionName(action: action),
+                    action: {
+                      viewModel.performAction(action, on: node)
+                    })
+                }
```
Comment (TheInkedEngineer) [inline Sources/App/Views/Sections/Editor/SourceTree/SourceTree.swift:39]:
```suggestion
                      viewModel.performAction(action, on: node)
                    }
                  )
                }
```

File: Sources/App/Views/Sections/Editor/SourceTree/SourceTreeViewModel.swift [hunk 5]
```diff
@@ -49,9 +52,25 @@ final class SourceTreeViewModel: ObservableObject {
...
+
     guard let node = node else {
-      // Case when we tap the `add new` button.
-      return EditorDetailViewModel(mode: .create)
+      return EditorDetailViewModel(
+        mode: .create,
+        onCancel: { [weak self] in
+          self?.isShowingCreateRequestDetailView = false
+        })
     }
```
Comment (TheInkedEngineer) [inline Sources/App/Views/Sections/Editor/SourceTree/SourceTreeViewModel.swift:74]:
```suggestion
          self?.isShowingCreateRequestDetailView = false
        }
      )
    }
```

File: Sources/App/Views/Sections/Editor/SourceTree/SourceTreeViewModel.swift [hunk 6]
```diff
@@ -65,9 +84,11 @@ final class SourceTreeViewModel: ObservableObject {
...
 
-      return EditorDetailViewModel(requestFile: node, requestFolder: requestFolderNode, requestParentFolder: parent) { [weak self] in
-        try? self?.refreshContent()
-      }
+      return EditorDetailViewModel(
+        requestFile: node, requestFolder: requestFolderNode, requestParentFolder: parent,
+        onSave: { [weak self] in
+          try? self?.refreshContent()
+        })
     }
```
Comment (TheInkedEngineer) [inline Sources/App/Views/Sections/Editor/SourceTree/SourceTreeViewModel.swift:92]:
```suggestion
        }
      )
    }
```

File: Sources/App/Views/Sections/Editor/SourceTree/SourceTreeViewModel.swift [hunk 7]
```diff
@@ -79,4 +100,35 @@ final class SourceTreeViewModel: ObservableObject {
...
+    case .delete:
+      return "􀈓  Delete"
+
+    case .edit:
+      return "􀈋  Edit"
+    }
+  }
+
+  func performAction(_ action: FileSystemNode.Action, on node: FileSystemNode? = nil) {
+    selectedNode = node
```
Comment (TheInkedEngineer) [inline Sources/App/Views/Sections/Editor/SourceTree/SourceTreeViewModel.swift:121]: doc

**Review by TheInkedEngineer — CHANGES_REQUESTED** [review]

**Review by TheInkedEngineer — APPROVED** [review]

**Review by FabrizioBrancati — APPROVED** [review]
LGTM 👍

**Pull request merged**